libcaf_net/caf/net/stream_socket.hpp

```diff
@@ -18,8 +18,11 @@
 
 #pragma once
 
+#include "caf/expected.hpp"
```

Comment:
Unnecessary.

Reply:
There is a function here that has an expected type as a return value. Is the implicit include good?

Reply:
Declaring a function prototype always only requires forward declarations for return type and argument types.